### DISCUSSION TITLE
Get autocompletion for kubernetes

### DISCUSSION THREAD

**Discussion by dadav** [LazyVim/LazyVim#1621] [Q&A] [answered]
Hello,

in my old config I was able to use https://github.com/someone-stole-my-name/yaml-companion.nvim to get some autocompletion for kubernetes spec files.

I tried to add the plugin like this:

```lua
  {
    "someone-stole-my-name/yaml-companion.nvim",
    ft = { "yaml" },
    opts = {
      builtin_matchers = {
        kubernetes = { enabled = true },
      },
    },
    dependencies = {
      { "neovim/nvim-lspconfig" },
      { "nvim-lua/plenary.nvim" },
      { "nvim-telescope/telescope.nvim" },
    },
    config = function()
      require("telescope").load_extension("yaml_schema")
    end,
  },
```

Which loads the plugin successfully on yaml files, but absolutely nothing happens. I don't get any error and have no clue whats wrong. Also the cmd `:Telescope yaml_schema` does nothing.

Does anyone have an idea or some hint how to debug this?

**Comment by folke** [comment, marked as answer]
When specifying a `config` function, then `setup` is **NOT** called automatically.

You can change the code to:

```lua
 config = function(_, opts)
     require("yaml-companion").setup(opts)
      require("telescope").load_extension("yaml_schema")
    end,
```

Apart from that, there's also a LazyVim extra for `yaml` that also automatically loads yaml/json schemas

**Comment by dadav** [comment]
> When specifying a `config` function, then `setup` is **NOT** called automatically.
> 
> You can change the code to:
> 
> ```lua
>  config = function(_, opts)
>      require("yaml-companion").setup(opts)
>       require("telescope").load_extension("yaml_schema")
>     end,
> ```
> 
> Apart from that, there's also a LazyVim extra for `yaml` that also automatically loads yaml/json schemas

Hi folke,

thanks for your help, it's working now!

Working config:

```lua
  {
    "someone-stole-my-name/yaml-companion.nvim",
    ft = { "yaml" },
    opts = {
      builtin_matchers = {
        kubernetes = { enabled = true },
      },
    },
    dependencies = {
      { "neovim/nvim-lspconfig" },
      { "nvim-lua/plenary.nvim" },
      { "nvim-telescope/telescope.nvim" },
    },
    config = function(_, opts)
      local cfg = require("yaml-companion").setup(opts)
      require("lspconfig")["yamlls"].setup(cfg)
      require("telescope").load_extension("yaml_schema")
    end,
  },
```

Using just `{ import = "lazyvim.plugins.extras.lang.yaml" },` does not give me the desired auto-completion tho. Check this out:

with `lang.yaml`

![not-working](https://github.com/LazyVim/LazyVim/assets/33197631/aa064a58-4a03-447e-966c-f1b68060fc80)

with `yaml-companion`

![working](https://github.com/LazyVim/LazyVim/assets/33197631/81772402-e69f-4431-a997-58a5e64b0f36)

**Reply by pnedkov** [reply]
I need a little help.
I have created a file `~/.config/nvim/lua/plugins/yaml-companion.lua` with the following content as suggested by @dadav :
```lua
return {
  {
    "someone-stole-my-name/yaml-companion.nvim",
    ft = { "yaml" },
    opts = {
      builtin_matchers = {
        kubernetes = { enabled = true },
      },
    },
    dependencies = {
      { "neovim/nvim-lspconfig" },
      { "nvim-lua/plenary.nvim" },
      { "nvim-telescope/telescope.nvim" },
    },

    config = function(_, opts)
      local cfg = require("yaml-companion").setup(opts)
      require("lspconfig")["yamlls"].setup(cfg)
      require("telescope").load_extension("yaml_schema")
    end,
  },
}
```

And I am still getting these errors when I open a Kubernetes yaml file:
![image](https://github.com/LazyVim/LazyVim/assets/6212427/69607df7-b66e-44e9-b7f3-82b5afd70fce)

I tried with lang.yaml enabled and disabled but it makes no difference. Can anyone give me a hint what am I doing wrong? Thank you.

**Reply by Dogacel** [reply]
@pnedkov can you share the whole file?

**Reply by szechp** [reply]
hey, yaml companion seems broken lately, i cant get it to work, even with the proper setup. do you have any alternatives for that?
